assets/src/edit-story/components/library/libraryProvider.js

```diff
@@ -45,26 +46,37 @@ const PAGE_LAYOUTS = {
   id: 'pageLayouts',
 };
 
+const LAZY_TABS = [MEDIA3P.id, TEXT.id, PAGE_LAYOUTS.id];
+
 function LibraryProvider({ children }) {
   const initialTab = MEDIA.id;
   const [tab, setTab] = useState(initialTab);
   const [textSets, setTextSets] = useState({});
+  const renderedTabs = useRef({});
   const insertElement = useInsertElement();
   const { insertTextSet, insertTextSetByOffset } = useInsertTextSet();
 
   const { showElementsTab } = useFeatures();
 
-  // Order here is important, as it denotes the actual visual order of elements.
+  const renderEmptyPane = useCallback((id) => {
+    const EmptyPane = (props) => <SharedPane id={getPaneId(id)} {...props} />;
+    return EmptyPane;
+  }, []);
+
   const tabs = useMemo(
-    () => [
-      MEDIA,
-      MEDIA3P,
-      ...(tab === TEXT.id ? [TEXT] : [{ icon: TextIcon, id: 'text' }]),
-      SHAPES,
-      ...(showElementsTab ? [ELEMS] : []),
-      PAGE_LAYOUTS,
-    ],
-    [showElementsTab, tab]
+    // Order here is important, as it denotes the actual visual order of elements.
+    () =>
+      [MEDIA, MEDIA3P, TEXT, SHAPES, showElementsTab && ELEMS, PAGE_LAYOUTS]
+        .filter(Boolean)
+        .map(({ icon, Pane, id }) => {
+          const isLazyTab = LAZY_TABS.includes(id);
+          const isActiveTab = tab === id;
+          const hasBeenRendered = renderedTabs.current[id];
+          return isLazyTab && !isActiveTab && !hasBeenRendered
+            ? { id, icon, Pane: renderEmptyPane(id) }
+            : { id, icon, Pane };
```

Comment:
Excellent work here - just a minor suggestion to reduce duplicated code (and I **really** prefer positively named booleans):

```js
          const shouldRenderPane = !isLazyTab || isActiveTab || hasBeenRendered;
          return { id, icon, Pane: shouldRenderPane ? Pane : renderEmptyPane(id) };
```